src/android/aidl/birthday_service/src/client.rs

```diff
@@ -12,19 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// ANCHOR: main
 //! Birthday service.
+use com_example_birthdayservice::aidl::com::example::birthdayservice::BirthdayInfo::BirthdayInfo;
+use com_example_birthdayservice::aidl::com::example::birthdayservice::IBirthdayInfoProvider::{
+    BnBirthdayInfoProvider, IBirthdayInfoProvider,
+};
 use com_example_birthdayservice::aidl::com::example::birthdayservice::IBirthdayService::IBirthdayService;
-use com_example_birthdayservice::binder;
+use com_example_birthdayservice::binder::{self, BinderFeatures, ParcelFileDescriptor};
+use std::fs::File;
+use std::io::prelude::*;
 
+// ANCHOR: main
 const SERVICE_IDENTIFIER: &str = "birthdayservice";
 
-/// Connect to the BirthdayService.
-pub fn connect() -> Result<binder::Strong<dyn IBirthdayService>, binder::StatusCode>
-{
-    binder::get_interface(SERVICE_IDENTIFIER)
-}
-
 /// Call the birthday service.
 fn main() -> Result<(), binder::Status> {
```

Comment:
You're unwrapping the I/O errors but not the Binder ones? I'm thinking you could just remove the `Result` here and use `unwrap` everywhere.

An alternative that would also make things consistent but with more effort: declare a new error type that covers all possible kinds, e.g., using the `thiserror` crate.

Reply:
Actually I can use `Box<dyn Error>`, which allows us to unify all the error types without using `thiserror` or `anyhow` (which is the crate that I'd otherwise use for a binary like this).